pkg/operator/operator.go

```diff
@@ -491,8 +537,11 @@ func New(
 		return nil, fmt.Errorf("failed to create federate certificate controller: %w", err)
 	}
 
-	o.controllersToRunFunc = append(o.controllersToRunFunc, csrFederateController.Run, csrController.Run)
-
+	if o.metricsServerEnabled {
+		o.controllersToRunFunc = append(o.controllersToRunFunc, csrFederateController.Run, csrController.Run, csrMetricsServerController.Run)
+	} else {
```

Comment:
nit: to avoid the else, we can keep the old append and `append(o.controllersToRunFunc, csrMetricsServerController.Run)` if `o.metricsServerEnabled`